addons/web/static/src/js/views/kanban/kanban_model.js

```diff
@@ -154,18 +156,43 @@ var KanbanModel = BasicModel.extend({
         var result = this._super.apply(this, arguments);
         var dp = result && this.localData[result.id];
         if (dp) {
+            // Add tooltipData key
             if (dp.tooltipData) {
                 result.tooltipData = $.extend(true, {}, dp.tooltipData);
             }
-            if (dp.progressBarValues) {
-                result.progressBarValues = $.extend(true, {}, dp.progressBarValues);
-            }
+            // Add isGroupedByM2ONoColumn key
             if (dp.fields[dp.groupedBy[0]]) {
                 var groupedByM2O = dp.fields[dp.groupedBy[0]].type === 'many2one';
                 result.isGroupedByM2ONoColumn = !dp.data.length && groupedByM2O;
             } else {
                 result.isGroupedByM2ONoColumn = false;
             }
+            // Add progressBarValues, loadMoreCount and loadMoreOffset key
+            let loadMoreCount = result.count - result.data.length;
+            let loadMoreOffset = result.data.length;
+            if (dp.progressBarValues) {
+                result.progressBarValues = $.extend(true, {}, dp.progressBarValues);
+                if (dp.activeFilter && dp.activeFilter.value) {
+                    // A progressbar filter is in an activated state?
+                    // Then set loadMoreCount and loadMoreOffset accordingly
+                    const barValues = dp.progressBarValues;
+                    const barField = barValues.field;
+                    const definedBarFieldValues = Object.keys(barValues.colors);
+                    const filterValue = dp.activeFilter.value;
+
+                    const filteredRecordsTotal = barValues.counts[filterValue];
+                    const filteredRecords = [...result.data].filter(element => {
```

Comment:
why `[...result.data]` ?

Reply:
I wanted to have IntelliSense in my IDE for the filter method :-)